### PR TITLE
Question submission

### PR DESCRIPTION
**Pull request by NickA02**
# Question Submission
Hello everyone. This is the API Endpoint for submission handling and Judge0 connection! 

There is one new route:
**POST** submissions/submit

The request takes in the user token header as usual, alongside a JSON *Submission*, of the form
```json
{
  "file_contents": "string",
  "question_num": "string"
}
```
and returns a *ConsoleLog* Json of the following form.
```json
{
  "console_log": "string"
}
```
This API Route is only accessible for users with active tests.

# Important Proposition

Also, I propose that from now on we have to include a lil joke in each PR... Here's mine:

**Why do space shuttles rely on Unix?**
*you can’t open Windows under water.*

### CODE REVIEW

**Comment by ItIsAndrewL** [comment]
Agreed on the joke haha

File: backend/services/submissions.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return self.run_submission(submission.question_num, team.name, forScore=False)
+
+    def run_submission(
+        self, question_num: int, team_name: str, forScore: bool = False
+    ) -> ConsoleLog:
+        """Run a submission on an Autograder and return the console logs
+        Args:
+            question_num (int): The question number
+            team_name (str): The team name
+            forScore (bool): Whether the submission is for final grading or not
```
Comment (ItIsAndrewL) [inline backend/services/submissions.py:51]: Is this where you intend my part to come in?

**Review by ItIsAndrewL — APPROVED** [review]
LGTM!!

**Review by mattfutch01 — APPROVED** [review]
I like it! LGTM

File: backend/services/submissions.py (hunk 1 above)
Comment (NickA02) [reply backend/services/submissions.py:51]: Yup!

**Pull request merged**